### ISSUE TITLE
Force-complete no longer works in new agent (1.0.5)

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#5768]
Creating this issue now that I can collect facts. This workflow was moved to "force-complete" - via webpage - in testbed:
amaltaro_TaskChain_Multicore_Agent105_CDB16_HG1504_150409_002328_4571

and it
a) did not kill the pending jobs
b) is still creating new jobs

If I open GQ, I no longer see any elements in there. The same applies for the workqueue_inbox doc, it has been deleted.

Investigating WorkQueueManager in the agent, I see the following interesting logs:
```
2015-04-09 08:33:16,666:INFO:WorkQueue:Canceling work for workflow(s): set(['amaltaro_TaskChain_Multicore_Agent105_CDB16_HG1504_150409_002328_4571'])
2015-04-09 08:33:17,211:INFO:DashboardReporter:Handling 1 jobs
... more hundreds of lines reporting to dashboard ...
2015-04-09 08:35:35,624:ERROR:WorkQueue:Aborting amaltaro_TaskChain_Multicore_Agent105_CDB16_HG1504_150409_002328_4571 wmbs subscription failed: (DatabaseError) ORA-00060: deadlock detected while waiting for resource
 'UPDATE wmbs_job SET retry_count = :retry WHERE id = :job' {'job': 19458, 'retry': 99999}
2015-04-09 08:35:47,714:INFO:WorkQueueBackend:set replication from GQ to LQ
...
2015-04-09 08:45:49,643:INFO:WorkQueue:Waiting for parent queue to delete "amaltaro_TaskChain_Multicore_Agent105_CDB16_HG1504_150409_002328_4571"
```

It's not clear to me also why it tries to update retries for one specific job to 99999.

**Comment by amaltaro** [comment]
and for completeness, these blocks have been deleted from local workqueue and workqueue_inbox as well, though I can still list them in wmbs_workflow table. I guess there is no way to stop this wf :-)

**Comment by amaltaro** [comment]
Actually it does not happen only in the new agent, I found a similar wmbs error in the 1.0.0 as well.
And it's not easily reproducible.
What about the following fix: https://github.com/amaltaro/WMCore/commit/b52e0acc8db98293cd86f1090baf1e99a2cad9f7
@ticoann ?

**Comment by ticoann** [comment]
Hi Alan, it seems there are 4 things happen when killWorkflow calls,

https://github.com/amaltaro/WMCore/blob/b52e0acc8db98293cd86f1090baf1e99a2cad9f7/src/python/WMCore/WorkQueue/WorkQueue.py

1. kill BossAir jobs.
2. update couchdb status of the jobs
3. update the dashboard status of jobs.
4. change the wmbs db job status.

I am guessing the problem is in 2. But not completely sure. Was that the a lot of jobs needed to be killed?

I think it is safe to clean the db as your patch did if the exception happens 2 or 3 but we still make 4 happen.

Maybe we should patch https://github.com/dmwm/WMCore/pull/5682 first which might reduce the couch failure case?

**Comment by amaltaro** [comment]
From the error message, I'd say it fails in 4), when updating wmbs database.
I agree, I'll first apply #5682 to testbed and run a few tests over the night and then we discuss further this issue ;)

**Comment by ticoann** [comment]
Sorry, I didn't read the error message carefully. In that case, I don't know how to handle that. I think we should check whether it will cause any side effect when wmbs job status is not updated.

**Comment by amaltaro** [comment]
Fixed by #5800
